app/classifier/tasks/drawing/index.cjsx

```diff
@@ -134,3 +142,10 @@ module.exports = createReactClass
     if e.target.checked
       newAnnotation = Object.assign {}, @props.annotation, _toolIndex: toolIndex
       @props.onChange newAnnotation
+    
+  handleDeleteAll: (toolIndex, e) ->
+    @constructor.closeAllMarks @props.task, @props.annotation
+    filteredShapes = @props.annotation.value.filter (a) -> a.tool != toolIndex
+    newAnnotation = Object.assign {}, @props.annotation
+    newAnnotation.value = filteredShapes
```

Comment:
Fun semantic point, this could be written:

```
newAnnotation = Object.assign {}, @props.annotation, { value: filteredShapes } 
```

...since it assigns from left to right